### PR TITLE
📦  Added business and person reports

### PR DESCRIPTION
**Pull request by drbobbeaty**
In order to facilitate KYB/KYC through Persona, we needed to add some
reports to the client so that it can be used to get the Watchlist,
Adverse Media reports for businesses and persons, and then the Business
Lookup and the Person PEP. These are all key components - that need to
be properly interpreted to make sure the Business and Person are who
they say they are, and not a potential Bad Actor.

### CODE REVIEW

File: src/report/index.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  }> {
+    // now wait a bit to get it - as we want to be synchronous on this
+    let cnt = 0
+    let chk: any
+    while (cnt < 60) {
+      // wait a bit, as we know Persona will take a few sec...
+      await setTimeout(500)
+      chk = await this.byId(reportId)
+      if (chk?.success && chk?.report?.attributes?.completedAt) {
+        return { ...chk }
```
Comment (edisongiraldo) [inline src/report/index.ts]: Why not just:
```
return chk
```

File: src/report/business.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    details?: PersonaCallDetails,
+    included?: PersonaIncluded,
+  }> {
+    const { synchronous, ...query } = data
+    return await this.client.report.run(
+      {
+        data: {
+          attributes: {
+            report_template_id: 'rptp_bWqnhyyu18zP8BFR6N6bt8sL',
+            query,
```
Comment (edisongiraldo) [inline src/report/business.ts:127]: Looking to another functions with the same call, we need to expand the query object:
```
{
  ...query,
}
```
Right?

**Review by edisongiraldo — CHANGES_REQUESTED** [review]
A few suggestions.

File: src/report/index.ts (hunk 1 above)
Comment (drbobbeaty) [reply src/report/index.ts]: fair.

File: src/report/business.ts (hunk 2 above)
Comment (drbobbeaty) [reply src/report/business.ts:127]: Nope, you're not looking at the API docs... the calls are different.

**Comment by drbobbeaty** [comment]
Tests out OK

**Review by edisongiraldo — APPROVED** [review]
Looks good to me.

**Pull request merged**